packages/core/database/src/utils/identifiers/index.ts

```diff
@@ -21,25 +41,164 @@ type NameOptions = {
   maxLength?: number;
 };
 
-export const getNameFromTokens = (tokens: NameToken[], max: number = MAX_DB_IDENTIFIER_LENGTH) => {
-  const fullLength = tokens
+// returns a hash of length len
+export function createHash(data: string, len: number): string {
+  if (!_.isInteger(len) || len <= 0) {
+    throw new Error(`createHash length must be a positive integer, received ${len}`);
+  }
+
+  // TODO: shake256 is based on SHA-3 and is slow, we don't care about cryptographically secure, only uniqueness and speed
+  //       investigate alternatives before releasing this. But it is only run on startup, so it should be fine.
+  const hash = crypto.createHash('shake256', { outputLength: Math.ceil(len / 2) }).update(data);
```

Comment:
just so i understand, you're using `shake256` because you want to control the length specifically?

Reply:
Yeah, it's the only common hash that lets you choose the output length. But on a technical level, I have to admit I'm just trusting that it exists because it's better than taking some other hash and using the first X chars from it